### PR TITLE
added connection settings to get by Uri

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
added connection settings to get by Uri

### Issues Resolved
NA

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/connection_templates/data_source.tf [hunk 1]
```diff
@@ -8,10 +8,20 @@ provider "oneview" {
...
\ No newline at end of file
+}
+
+# Gets connection template data from uri
+data "oneview_connection_templates" "connectionTemplatesURI" {
+ uri = data.oneview_connection_templates.connectionTemplates.uri
+}
+
+output "oneview_connection_template_valueURI" {
+  value = data.oneview_connection_templates.connectionTemplates.name
```
Comment (yuvirani) [inline examples/connection_templates/data_source.tf:26]: it should be oneview_connection_templates.connectionTemplatesURI

File: oneview/data_source_connection_templates.go [hunk 2]
```diff
@@ -58,14 +60,15 @@ func dataSourceConnectionTemplates() *schema.Resource {
 			},
 			"name": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```
Comment (yuvirani) [inline oneview/data_source_connection_templates.go:63]: If name not given, we may get idempotency issues. Please verify and update as Optional and Computed.

File: oneview/resource_connection_templates.go [hunk 3]
```diff
@@ -102,7 +102,7 @@ func resourceConnectionTemplates() *schema.Resource {
 func resourceConnectionTemplatesRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 	id := d.Id()
-	cTemplate, err := config.ovClient.GetConnectionTemplateByName(id)
+	cTemplate, err := config.ovClient.GetConnectionTemplateByURI(utils.Nstring(id))
 	if err != nil || cTemplate.URI.IsNil() {
 		d.SetId("")
 		return nil
```
Comment (AvijitOS) [inline oneview/resource_connection_templates.go:108]: According to me resourceConnectionTemplatesUpdate have two issues:
1. it is trying to update all the properties(include in schema)...like state, status, modified, created..i think we should not raise put call for those
2. uri := strings.Split(updateOptions.URI.String(), "/")[3]
	template, err := config.ovClient.UpdateConnectionTemplate(uri, updateOptions)
  why we need to do split operation then getting the id and pass it to it? we can just pass the uri itself

**Review by AvijitOS — COMMENTED** [review]
we will discuss over the call

File: docs/d/connection_templates.html.markdown [hunk 4]
```diff
@@ -18,13 +18,24 @@ data "oneview_connection_templates" "test" {
...
+}
+```
+
+```hcl
+data "oneview_connection_templates" "testingViaUri" {
+ uri = "/rest/connection-templates/defaultConnectionTemplate" 
+}
+
+output "oneview_connection_templates_value" {
+ value = data.oneview_connection_templates.test.uri
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply docs/d/connection_templates.html.markdown:31]: done

File: oneview/data_source_connection_templates.go (hunk 2 above)
Comment (AsisBagga) [reply oneview/data_source_connection_templates.go:63]: done

File: oneview/resource_connection_templates.go (hunk 3 above)
Comment (AsisBagga) [reply oneview/resource_connection_templates.go:101]: done

**Review by yuvirani — APPROVED** [review]

**Review by AvijitOS — APPROVED** [review]

**Pull request merged**